### PR TITLE
Fix windows encoding issues

### PR DESCRIPTION
**Pull request by peternied**
### Description
Fix windows encoding issues

Adds spotbugs [1] to detect internationalization before they are added to the codebase, also fixed several encoding bugs that impact windows users.

[1] https://spotbugs.readthedocs.io/en/stable/index.html

### Issues Resolved
- Resolves https://github.com/opensearch-project/security/issues/2194

### Check List
- [X] New functionality includes testing
- [ ] New functionality has been documented
- [X] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -34,6 +34,7 @@ jobs:
           -x spotlessCheck
           -x checkstyleMain
           -x checkstyleTest
+          -x spotbugsMain
```
Comment (scrawfor99) [inline .github/workflows/ci.yml:37]: It looks like this workflow may be causing the plugin to not install properly? Is this just a lint error or something with this specific workflow?

File: spotbugs-include.xml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+<FindBugsFilter>
+    <Match>
+        <Bug category="I18N" />
+    </Match>
+</FindBugsFilter>
```
Comment (scrawfor99) [inline spotbugs-include.xml:5]: Do you think this should be added as a separate PR or is fine to include it in this one even though it seems not directly related to the parsing issue--maybe it is and I am wrong?

File: src/main/java/org/opensearch/security/auditlog/impl/AbstractAuditLog.java [hunk 3]
```diff
@@ -419,7 +419,7 @@ public void logDocumentWritten(ShardId shardId, GetResult originalResult, Index
                     try (XContentParser parser = XContentHelper.createParser(NamedXContentRegistry.EMPTY, THROW_UNSUPPORTED_OPERATION, originalResult.internalSourceRef(), XContentType.JSON)) {
                         Object base64 = parser.map().values().iterator().next();
                         if(base64 instanceof String) {
-                            originalSource = (new String(BaseEncoding.base64().decode((String) base64)));
+                            originalSource = (new String(BaseEncoding.base64().decode((String) base64), StandardCharsets.UTF_8));
```
Comment (scrawfor99) [inline src/main/java/org/opensearch/security/auditlog/impl/AbstractAuditLog.java:422]: Since we use this in numerous places and I believe there used to be a content type setting in a separate file, do we want to make a setting for the encoding method or does hard coding make more sense because we are setting it to the most broad option?

File: spotbugs-include.xml (hunk 2 above)
Comment (cwperks) [reply spotbugs-include.xml:5]: This is helpful to find instances where the code is relying on the default encoding. See https://spotbugs.readthedocs.io/en/latest/bugDescriptions.html?highlight=i18n#dm-reliance-on-default-encoding-dm-default-encoding

**Review by cwperks — APPROVED** [review]

File: spotbugs-include.xml (hunk 2 above)
Comment (peternied) [reply spotbugs-include.xml]: I did not add more rules because we are in violation of _many_ different rules, but I didn't see any types that would create platform issues.

If you'd like to see the full list, delete line 3 and run the gradle task, you'll see hundreds of things!

File: src/main/java/org/opensearch/security/auditlog/impl/AbstractAuditLog.java (hunk 3 above)
Comment (peternied) [reply src/main/java/org/opensearch/security/auditlog/impl/AbstractAuditLog.java:422]: When we were on unix environments the setting value was always UTF-8 both inside of security and OpenSearch so everything was happy.  Since OpenSearch widely forces the use of UTF-8 and security does in some spots this was the source of our issues.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (peternied) [reply .github/workflows/ci.yml:37]: The broken workflow was because of the linelinter, fix now!

**Comment by peternied** [comment]
@cwperks @DarshitChanpura Ready for review, thanks!

**Review by cwperks — APPROVED** [review]

**Comment by peternied** [comment]
@opensearch-project/security Can I get another review on this please?

**Comment by peternied** [comment]
Manual backports created:
- https://github.com/opensearch-project/security/pull/2217
- https://github.com/opensearch-project/security/pull/2218
